src/vmm/src/builder.rs

```diff
@@ -315,6 +360,20 @@ fn create_vmm_and_vcpus(
     Ok((vmm, vcpus))
 }
 
+#[cfg(target_arch = "x86_64")]
+#[allow(unused)]
+fn get_msr_values(_vcpu: Option<&Vcpu>, _template: &X86_64CpuTemplate) -> HashMap<u32, u64> {
+    // TODO - Use a created vCPU to retrieve MSR values
+    Default::default()
+}
+
+#[cfg(target_arch = "aarch64")]
+#[allow(unused)]
+fn get_reg_values(_vcpu: Option<&Vcpu>, _template: &Aarch64CpuTemplate) -> HashMap<u64, u128> {
+    // TODO - Use a created vCPU to retrieve register values
+    Default::default()
+}
```

Comment:
Why `Option<&Vcpu>`? How can you retrieve anything if you don't have vcpu?

Reply:
This is a fair question. So the idea here was to be more flexible, in allowing use to just write whole register values via the template, in which case when the bitmap is applied we can just use the value from the map. I'm on the fence about being more rigorous in requiring the lookup for a bitmap to be usable, but I don't have a strong preference at this stage. Wdy?

Reply:
@mattschlebusch could you elaborate on your point? Do you mean, in a very special case where the bitmap doesn't have any wildcards ("x"), we don't need any KVM calls and can build the output directly from the template? Do you think this warrants making Vcpu optional? Where would the logic deciding on pass vs not pass the real Vcpu reside?